Deliverance/DeliveranceMailChimpList.php

```diff
@@ -348,66 +168,46 @@ protected function initListMergeArrayMap()
 	// subscriber methods
 	// {{{ public function subscribe()
 
-	public function subscribe(
-		$address,
-		array $info = array(),
-		$send_welcome = true,
-		array $array_map = array()
-	) {
+	public function subscribe($address, array $info = array())
+	{
 		$result = false;
 		$queue_request = false;
 
 		if ($this->isAvailable()) {
-			$merges     = $this->mergeInfo($info, $array_map);
-			$parameters = array(
-				$this->shortname,
-				$address,
-				$merges,
-				$this->email_type,
-				$this->double_opt_in,
-				$this->update_existing,
-				$this->replace_interests,
-				$send_welcome,
-				);
+			$merges = $this->mergeInfo($info);
+			$interests = $this->interestInfo($info);
 
 			try {
 				$result = $this->callClientMethod(
-					'listSubscribe',
-					$parameters);
-			} catch (DeliveranceAPIConnectionException $e) {
-				// exception is known, API is not available.
+					'PUT',
+					sprintf(
+						'lists/%s/members/%s',
+						$this->shortname,
+						$this->client->subscriberHash($address)
+					),
+					[
+						'email_address' => $address,
+						'email_type' => $this->email_type,
+						'status' => 'subscribed',
+						'merge_fields' => $merges,
+						'interests' => $interests
+					]
+				);
+			} catch (DeliveranceMailChimpTimeoutException $e) {
 				$queue_request = true;
-			} catch (DeliveranceException $e) {
-				// gracefully handle exceptions that we can provide nice
-				// feedback about.
-				switch ($e->getCode()) {
-				case self::INVALID_ADDRESS_ERROR_CODE:
-					$result = DeliveranceList::INVALID;
-					break;
-
-				case self::BANNED_ADDRESS_ERROR_CODE:
-					// log these to keep track of how frequent they are. If they
-					// become frequent we should build a better message for the
-					// user.
-					$e = new DeliveranceBannedAddressException($e);
-					$e->processAndContinue();
-
-					$result = DeliveranceList::INVALID;
-					break;
-
-				default:
-					throw $e;
-				}
+			} catch (DeliveranceMailChimpServerException $e) {
+				$queue_request = true;
+			} catch (DeliveranceMailChimpClientException $e) {
+				$result = DeliveranceList::INVALID;
```

Comment:
While testing, I got this error for other reasons than having an invalid email. Not sure if this is something we can check for or if it's worth checking for right now. I got the error for trying to submit invalid interests.